src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx

```diff
@@ -99,97 +60,85 @@ export const PupilViewsProgrammeListing = ({
   }
 
-  const backlink = resolveOakHref({
-    page: "pupil-subject-index",
-    yearSlug,
-  });
-
   const subjectSlug = programmes[0]?.programmeFields.subjectSlug;
   const subjectDescription = programmes[0]?.programmeFields.subject;
   const yearDescriptions = programmes[0]?.programmeFields.yearDescription;
 
-  // TODO - switch statement can be refactored
   const topNavSlot = () => {
-    switch (true) {
-      // examboard is chosen and there are multiple tiers
-      case chosenExamboard !== null && tiers.length > 1:
-        return (
-          <OakTertiaryButton
-            iconName="arrow-left"
-            onClick={() => setChosenExamboard(null)}
-          >
-            Change examboard
-          </OakTertiaryButton>
-        );
-      default:
-        return (
-          <OakTertiaryButton iconName="arrow-left" element="a" href={backlink}>
-            Change subject
-          </OakTertiaryButton>
-        );
+    if (chosenExamboard !== null && tiers.length > 1) {
+      return (
+        <OakTertiaryButton
+          iconName="arrow-left"
+          onClick={() => setChosenExamboard(null)}
+        >
+          Change examboard
+        </OakTertiaryButton>
+      );
+    } else {
+      return (
+        <OakTertiaryButton
+          iconName="arrow-left"
+          element="a"
+          href={resolveOakHref({
+            page: "pupil-subject-index",
+            yearSlug,
+          })}
+        >
+          Change subject
+        </OakTertiaryButton>
+      );
     }
   };
 
-  const breadcrumbs = () => {
+  const breadcrumbs: string[] = [yearDescriptions];
+  if (chosenExamboard?.examboard && tiers.length > 1) {
+    breadcrumbs.push(chosenExamboard.examboard);
+  }
+
+  const BrowseOptions = () => {
     if (
-      !chosenExamboard ||
-      tiers.length <= 1 ||
-      chosenExamboard.examboard === null
+      (chosenExamboard !== null && tiers.length > 1) ||
+      (examboards.length <= 1 && tiers.length >= 1)
     ) {
-      return [yearDescriptions];
+      return (
+        <BrowseTierSelector
+          tiers={tiers}
+          baseSlug={baseSlug}
+          examboardSlug={
+            chosenExamboard ? chosenExamboard.examboardSlug : undefined
+          }
+          phaseSlug={phaseSlug}
+        />
+      );
+    } else if (examboards.length > 1 && chosenExamboard === null) {
+      return (
+        <BrowseExamboardSelector
+          examboards={examboards}
+          baseSlug={baseSlug}
+          onClick={
+            tiers.length > 1
+              ? (examboard) => setChosenExamboard(examboard)
+              : undefined
+          }
+          phaseSlug={phaseSlug}
+        />
+      );
+    } else {
+      return <div>No programme factors to be selected</div>;
     }
-    return [yearDescriptions, chosenExamboard.examboard] as string[];
-  };
-
-  // TODO - Immediate function and switch statement can be refactored
-  const BrowseOptions = () => {
-    return (() => {
-      switch (true) {
```

Comment:
Replaced `switch` with `if` statement as I think it’s a bit clearer when there are only 2 options.